packages/wrangler/src/index.tsx

```diff
@@ -72,10 +72,14 @@ async function readConfig(configPath?: string): Promise<Config> {
     "usage_model",
   ];
 
-  Object.keys(config.env || {}).forEach((env) => {
+  const environments = config.env ?? {};
+  Object.values(environments).forEach((env) => {
+    if (env === undefined) {
+      return;
```

Comment:
I don't think we want to return here, instead we'll have to create the object and copy the inherited fields on to it.

Reply:
The previous code would crash if the value of `config.env[key]` was undefined, since it had the following unguarded expression: `config.env[env]` (where in that case `env` is the key.